service/src/main/java/bio/terra/workspace/service/resource/referenced/cloud/any/gitrepo/ReferencedGitRepoHandler.java

```diff
@@ -19,4 +22,9 @@ public static ReferencedGitRepoHandler getHandler() {
   public WsmResource makeResourceFromDb(DbResource dbResource) {
     return new ReferencedGitRepoResource(dbResource);
   }
+
+  public String generateCloudName(@Nullable UUID workspaceUuid, String resourceName) {
+    throw new FeatureNotSupportedException(
```

Comment:
For referenced resources we'll never implement this method - imo a 400 would be more accurate than a 501. Could you change the exception thrown, at least for the referenced resources?

Reply:
Wow. I just find the FeatureNotSupportedException belongs to 501. Find the BadRequestExcpetion belongs to 400. Taken notes on it! Thank you Zach.